File: Cpp/1685_Sum_of_Absolute_Differences_in_a_Sorted_Array.cpp
class Solution {
public:
    vector<int> getSumAbsoluteDifferences(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(n, 0);
        vector<int> prefix(n, 0);
        prefix[0] = nums[0];
        
        for(int i = 1; i < n; ++i) {
            prefix[i] = prefix[i-1] + nums[i];
        }
        
        for(int i = 0; i < n; ++i) {
            int left = i > 0 ? i * nums[i] - prefix[i - 1] : 0;
            int right = prefix[n - 1] - prefix[i] - (n - 1 - i) * nums[i];
            
            result[i] = left + right;
        }
        
        return result;
    }
};

File: Kotlin/1685_Sum_of_Absolute_Differences_in_a_Sorted_Array.kt
class Solution {
    fun getSumAbsoluteDifferences(nums: IntArray): IntArray {
        val n = nums.size
        val result = IntArray(n)
        val prefix = IntArray(n)
        prefix[0] = nums[0]
        
        for(i in 1 until n) {
            prefix[i] = prefix[i - 1] + nums[i]
        }
        
        for (i in 0 until n) {
            val left = if(i > 0) i * nums[i] - prefix[i - 1] else 0
            val right = prefix[n - 1] - prefix[i] - (n - 1 - i) * nums[i]
            
            result[i] = left + right
        }
        
        return result
    }
}